File: src/com/williambarkoff/csx/rustremoval/RandomWalkApp.java
package com.williambarkoff.csx.rustremoval;

import org.opensourcephysics.controls.AbstractSimulation;
import org.opensourcephysics.controls.SimulationControl;
import org.opensourcephysics.display.Circle;
import org.opensourcephysics.frames.PlotFrame;

import java.util.ArrayList;
import java.util.Random;

public class RandomWalkApp extends AbstractSimulation {
    private PlotFrame plotFrame = new PlotFrame("x", "y", "Random Walk");
    private ArrayList<MovingCircle> circles = new ArrayList<MovingCircle>();

    public static void main(String[] args) {
        SimulationControl.createApp(new RandomWalkApp());
    }

    @Override
    public void doStep() {
        circles.forEach(MovingCircle::move);
    }

    @Override
    public void initialize() {
        for(int i = 1; i <= 50; i++) {
            MovingCircle circle = new MovingCircle();
            circles.add(circle);
            plotFrame.addDrawable(circle);
        }
        plotFrame.setPreferredMinMax(-25.0, 25.0, -25.0, 25.0);
    }


    private class MovingCircle extends Circle {
        Random gen = new Random();

        public void move() {
            int randX = gen.nextInt(2);
            if(randX == 0) {
                this.x--;
            } else if(randX == 1) {
                this.x++;
            }

            int randY = gen.nextInt(2);
            if(randY == 0) {
                this.y--;
            } else if(randY == 1) {
                this.y++;
            }
        }
    }
}

File: src/com/williambarkoff/csx/rustremoval/RandomWalkApp.kt
package com.williambarkoff.csx.rustremoval

import org.opensourcephysics.controls.AbstractSimulation
import org.opensourcephysics.controls.SimulationControl
import org.opensourcephysics.display.Circle
import org.opensourcephysics.frames.PlotFrame
import kotlin.random.Random

fun main() {
    SimulationControl.createApp(RandomWalkApp())
}

class RandomWalkApp : AbstractSimulation() {
    internal var plotFrame = PlotFrame("x", "y", "Random Walk")
    internal var circles = ArrayList<MovingCircle>()

    override fun doStep() {
        circles.forEach { it.move() }
    }

    override fun initialize() {
        for(i in 1..50) {
            val circle = MovingCircle()
            circles.add(circle)
            plotFrame.addDrawable(circle)
        }
        plotFrame.setPreferredMinMax(-25.0, 25.0, -25.0, 25.0)
    }
}

class MovingCircle: Circle() {
    fun move() {
        val randX = Random.nextInt(0, 2)
        when(randX) {
            0 -> this.x--
            1 -> this.x++
        }

        val randY = Random.nextInt(0, 2)
        when(randY) {
            0 -> this.y--
            1 -> this.y++
        }

    }
}